presto-tests/src/test/java/io/prestosql/connector/system/TestSystemConnector.java

```diff
@@ -135,20 +136,21 @@ public void testQueryDuringAnalysisIsCaptured()
             getQueryRunner().execute(format("EXPLAIN SELECT 1 AS %s FROM test_table", testQueryId));
         });
 
-        Thread.sleep(100);
-
-        assertQuery(
+        assertQueryEventually(
+                getSession(),
                 format("SELECT state FROM system.runtime.queries WHERE query LIKE '%%%s%%' AND query NOT LIKE '%%system.runtime.queries%%'", testQueryId),
-                "VALUES 'WAITING_FOR_RESOURCES'");
+                "VALUES 'WAITING_FOR_RESOURCES'",
+                new Duration(10, SECONDS));
         assertFalse(metadataFuture.isDone());
         assertFalse(queryFuture.isDone());
 
         metadataFuture.set(CONNECTOR_TABLE_METADATA);
-        Thread.sleep(100);
 
-        assertQuery(
+        assertQueryEventually(
+                getSession(),
                 format("SELECT state FROM system.runtime.queries WHERE query LIKE '%%%s%%' AND query NOT LIKE '%%system.runtime.queries%%'", testQueryId),
-                "VALUES 'FINISHED'");
+                "VALUES 'FINISHED'",
+                new Duration(10, SECONDS));
         assertTrue(queryFuture.isDone());
     }
 
```

Comment:
let's use `assertQueryEventually` in `testQueryKillingDuringAnalysis`

Reply:
it's not applicable there